runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -262,7 +342,13 @@ private void insertAllDependingFileDescriptorProtos(TableDescriptors.Builder tab
             }
 
             // Add the fileDescriptorProto into the tableDescriptor map.
-            tableDescriptorsBuilder.putFileDescriptors(fileDescriptorProto.getName(), fileDescriptorProto);
+            tableDescriptorsBuilder.putFileDescriptors(fileDescriptorProto.getName(),
+                    FileDescriptorProto.getDefaultInstance());
+
+            // Add the actual descriptor into a common pool of descriptors to avoid duplication
+            ProtobufFileName protoFileName = ProtobufFileName.newBuilder().setFileName(fileDescriptorProto.getName()).build();
+            ProtobufFileDescriptor protoFd = ProtobufFileDescriptor.newBuilder().setFileDescriptor(fileDescriptorProto).build();
+            allDescriptors.putIfAbsent(protoFileName, new CorfuRecord<>(protoFd, null));
```

Comment:
Can you do a sanity check before you insert where the protoFileName actually maps to the same proto file descriptor value? otherwise this could fail silently and over write different descriptors.

Reply:
yes we do that check, please see the method recordNewSchema() this is just placing the descriptor in a stack local map.